profile/store.go

```diff
@@ -25,6 +26,8 @@ type Store interface {
 	Owner() *Profile
 	// SetOwner handles updates to the current user profile at runtime
 	SetOwner(own *Profile) error
+	// Active is the active profile that represents the current user
+	Active(ctx context.Context) *Profile
```

Comment:
I'm assuming a follow-up will define something a lot like [this](https://github.com/qri-io/qri/blob/63fadf8339de9a603aff382f95bac89b92b0361b/remote/context.go#L1-L26) in the `key` package, adding a `context.go` file & functions for adding & getting key identifier (and eventually public key) from a context, and that this `Active` function will effectively be:

```go
func ActiveProfile(ctx context.Context) (*Profile, error) {
  keyID, err := key.KeyIDFromContext(ctx)
  return store.GetByKeyID(keyID)
}
```

If so, let's make `ActiveProfile` a function that accepts a store, reducing the need for implementations to add boilerplate:

```go
package profile

func ActiveProfile(ctx context.Context, store Store) (*Profile, error) {
  keyID, err := key.KeyIDFromContext(ctx)
  return store.GetByKeyID(keyID)
}
```

When embedding values in contexts, it's best to keep the context accessors high up the call stack. I'd personally love to see pseudocode like this:

```go
package lib

func (m *DatasetMethods) Save(ctx context.Context, params *SaveParams) (*dataset.Dataset, error) {

  keyID, privKey, err := key.PrivateKeyFromContext(ctx, m.Keystore)
  // error earliest if we don't have the private key
  
  // this API isn't worked out yet, but something like...
  cred, err := access.CredentialTokenForResource(p.DatasetID, keyID)
  if err != nil {
    return ErrAccessDenied
  }

  return base.Save(ctx, params, privKey /* ... */)
}
```

For the next PR if you haven't already had a read through the golang blog post on context, def worth starting there: https://blog.golang.org/context

Reply:
Yes, you're pretty close to the target on what the goal is.
There is one key distinction I want to throw out there:
- There is managing multiple keys for a profile
- And then there is managing multiple profiles with their keys

We technically need to cover for both. I will first be covering the multiple profiles path as we don't have the multiple key stuff sorted yet.